packages/react-core/src/helpers/Popper/FindRefWrapper.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+interface FindRefWrapperProps {
+  children: React.ReactNode;
+  onFoundRef: any;
+}
+
+/**
+ * This component wraps any ReactNode and finds its ref
+ * It has to be a class for findDOMNode to work
+ * Ideally, all components used as triggers/toggles are either:
+ * - class based components we can assign our own ref to
+ * - functional components that have forwardRef implemented
+ * However, there is no guarantee that is what will get passed in as trigger/toggle in the case of tooltips and popovers
+ */
+export class FindRefWrapper extends React.Component<FindRefWrapperProps> {
+  static displayName = 'FindRefWrapper';
+  componentDidMount() {
+    // eslint-disable-next-line react/no-find-dom-node
+    const root = ReactDOM.findDOMNode(this);
+    this.props.onFoundRef(root);
```

Comment:
Why do we need this component? Why not add a `componentDidMount` to `Popper.tsx`?

Reply:
I need to find 2 different refs for the toggle and the menu, it's cleaner to have this reusable class component, also Popper gets to be a functional components and i can use hooks such as the `usePopper` hook

Reply:
Fair enough. There are conflicts now after merging #4308